File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/GyroMecanum/src/Robot.cpp
#include <AnalogGyro.h>
#include <IterativeRobot.h>
#include <Joystick.h>
#include <RobotDrive.h>

/**
 * This is a sample program that uses mecanum drive with a gyro sensor to
 * maintian rotation vectorsin relation to the starting orientation of the robot
 * (field-oriented controls).
 */
class Robot: public IterativeRobot {
public:
	void RobotInit() {
		// invert the left side motors
		// you may need to change or remove this to match your robot
		myRobot.SetInvertedMotor(RobotDrive::kFrontLeftMotor, true);
		myRobot.SetInvertedMotor(RobotDrive::kRearLeftMotor, true);

		gyro.SetSensitivity(kVoltsPerDegreePerSecond);
	}

	/**
	 * Mecanum drive is used with the gyro angle as an input.
	 */
	void TeleopPeriodic() {
		myRobot.MecanumDrive_Cartesian(joystick.GetX(), joystick.GetY(),
				joystick.GetZ(), gyro.GetAngle());
	}

private:
	// Gyro calibration constant, may need to be adjusted
	// Gyro value of 360 is set to correspond to one full revolution
	static constexpr double kVoltsPerDegreePerSecond = 0.0128;

	RobotDrive myRobot { 0, 1, 2, 3 };
	AnalogGyro gyro { 0 };
	Joystick joystick { 0 };
};

START_ROBOT_CLASS(Robot)

File: edu.wpi.first.wpilib.plugins.java/resources/templates/examples/GyroMecanum/src/org/usfirst/frc/team190/robot/Robot.java
package $package;

import edu.wpi.first.wpilibj.AnalogGyro;
import edu.wpi.first.wpilibj.IterativeRobot;
import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.RobotDrive;
import edu.wpi.first.wpilibj.RobotDrive.MotorType;

/**
 * This is a sample program that uses mecanum drive with a gyro sensor to
 * maintian rotation vectorsin relation to the starting orientation of the robot
 * (field-oriented controls).
 */
public class Robot extends IterativeRobot {

    // gyro calibration constant, may need to be adjusted;
    // gyro value of 360 is set to correspond to one full revolution
    private static final double kVoltsPerDegreePerSecond = 0.0128;

    private static final int kFrontLeftMotorPort = 0;
    private static final int kFrontRightMotorPort = 1;
    private static final int kRearLeftMotorPort = 2;
    private static final int kRearRightMotorPort = 3;
    private static final int kGyroPort = 0;
    private static final int kJoystickPort = 0;

    private RobotDrive myRobot;
    private AnalogGyro gyro;
    private Joystick joystick;

    public void robotInit() {
        myRobot = new RobotDrive(kFrontLeftMotorPort, kFrontRightMotorPort,
            kRearLeftMotorPort, kRearRightMotorPort);
        gyro = new AnalogGyro(kGyroPort);
        joystick = new Joystick(kJoystickPort);

        // invert the left side motors
        // you may need to change or remove this to match your robot
        myRobot.setInvertedMotor(MotorType.kFrontLeft, true);
        myRobot.setInvertedMotor(MotorType.kRearLeft, true);

        gyro.setSensitivity(kVoltsPerDegreePerSecond);
    }

    /**
     * Mecanum drive is used with the gyro angle as an input.
     */
    public void teleopPeriodic() {
        myRobot.mecanumDrive_Cartesian(joystick.getX(), joystick.getY(), joystick.getZ(), gyro.getAngle());
    }

}
